### DISCUSSION TITLE
how do I upgrade from v3.15.2  to v3.23(Dec 2021)

### DISCUSSION THREAD

**Discussion by hsamdani12** [microsoft-search/pnp-modern-search#2633] [Q&A] [answered]
Hi,

We have an old version of PnP Search Installed - v3.15.2 (pnp-react-search-refiners.sppkg) - installed in the tenant app catalog  to v3.23 (pnp-modern-search-parts.sppkg). The sppkg have different names. Can someone guide what needs to be done ?
I don't want to break the existing PnP webparts deployed.

**Comment by wobba** [comment, marked as answer]
1. Install 3.15 in a test tenant
2. Configure up a page
3. Remove 3.15, and install 3.23
4. Verify if the page works

If it doesn't work, then you have to reconfigure the pages. As I don't recall exactly if there are breaking changes between 3.15 and 3.23 (you can read the release notes on all versions to check).

The better option is v4, but that will require re-setup of all pages and new webparts.

**Comment by hsamdani12** [comment]
@wobba. thanks. Yes going for v4 and resetting the webparts.
